File: mockez-back/src/main/java/com/mockez/domain/model/entity/GroupAccess.java
package com.mockez.domain.model.entity;

import com.mockez.domain.model.entity.composite_key.GroupAccessPK;
import com.mockez.domain.model.entity.enumeration.Role;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

import javax.persistence.Column;
import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.ForeignKey;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "group_access")
@ToString(callSuper = true)
@SuperBuilder(toBuilder = true)
public class GroupAccess extends Base {

    @EmbeddedId
    private GroupAccessPK id;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Role role = Role.GROUP_ASSOCIATE;

    @ManyToOne
    @ToString.Exclude
    @JoinColumn(nullable = false, foreignKey = @ForeignKey(name = "user_group_access_fk"), insertable=false, updatable=false)
    private User user = User.builder().build();

    @ManyToOne
    @ToString.Exclude
    @JoinColumn(nullable = false, foreignKey = @ForeignKey(name = "group_group_access_fk"),  insertable=false, updatable=false)
    private Group group = Group.builder().build();
}

File: mockez-front/src/app/core/model/group-access.ts
import { Base } from '@core/model/base';
import { GroupAccessPK } from '@core/model/composite_key/group-access-pk';
import { Role } from '@core/model/enumeration/role';
import { User } from '@core/model/user';
import { Group } from '@core/model/group';

export class GroupAccess extends Base {

  private _id: GroupAccessPK | undefined = undefined;
  private _role: Role | undefined = undefined;
  private _user: User | undefined = undefined;
  private _group: Group | undefined = undefined;

  get id(): GroupAccessPK | undefined {
    return this._id;
  }

  set id(value: GroupAccessPK | undefined) {
    this._id = value;
  }

  get role(): Role | undefined {
    return this._role;
  }

  set role(value: Role | undefined) {
    this._role = value;
  }

  get user(): User | undefined {
    return this._user;
  }

  set user(value: User | undefined) {
    this._user = value;
  }

  get group(): Group | undefined {
    return this._group;
  }

  set group(value: Group | undefined) {
    this._group = value;
  }
}
